python/ray/tests/test_actor_failures.py

```diff
@@ -989,6 +998,45 @@ def verify():
     wait_for_condition(verify)
 
 
+def test_exit_actor_async_actor_nested_task(shutdown_only, tmp_path):
+    async_temp_file = tmp_path / "async_actor.log"
+    async_temp_file.touch()
+
+    @ray.remote
+    class AsyncActor:
+        def __init__(self):
+            def f():
+                print("atexit handler")
+                with open(async_temp_file, "w") as f:
+                    f.write("Async Actor\n")
+
+            atexit.register(f)
+
+        async def start_exit_task(self):
+            asyncio.create_task(self.exit())
+
+        async def exit(self):
+            exit_actor()
+
+    a = AsyncActor.remote()
+    ray.get(a.__ray_ready__.remote())
+    with pytest.raises(ray.exceptions.RayActorError) as exc_info:
+        ray.get(a.start_exit_task.remote())
```

Comment:
hm this test doesn't look correct. the method starts a background task so there is no guarantee that runs before the method finishes successfully. I think what you need to do is start the background task and have it block on a `SignalActor` before it actually calls `exit_actor()`, check that `start_exit_task` completes successfully, then unblock the background task so it calls `exit_actor()`, then use `wait_for_condition` on `ray.get(a.noop_method.remote())` to check that eventually subsequent methods will return the actor error